modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/modal/Modal.js

```diff
@@ -78,7 +78,7 @@ function Modal({
 				setTitle(data.title);
 			}
 
-			if (data.disableHeader) {
+			if (data.disableHeader !== undefined) {
```

Comment:
Would it maybe make sense to be opinionated here, something like
```js
setDisableHeader(data.disableHeader ?? true);
```

With the code in PR, `undefined` will leave the previous state, whatever it was. Do we want this?

Reply:
@markocikos Need to revert it back. Updating the state as you propose prevents the modal header to show up

Reply:
Don't we want to hide it by default? So that we don't need to migrate commerce solutions. It would mean that we need to set `disableHeader` to false when it comes from DSM. Is that doable?

Reply:
Yes, we want it to hide it by default. This is why we [initialize](https://github.com/liferay-frontend/liferay-portal/pull/4196/files#diff-d765b3175e62a32e65803070a377d44fffcea8a775a09050388f603be1a96f8fR36) the `disableHeader` state to `true`.
The problem (no modal title at all) appears when we reset the state.

Reply:
Can we set `data.disableHeader` to `false` when it comes from DSM?

Reply:
Issue found. Wrapping the state setter within the if block was causing the problem :sweat: